src/app.md

```diff
@@ -189,6 +190,69 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-tar-iterate]: #ex-tar-iterate
+<a name="ex-tar-iterate"></a>
+## Iterate through the entries of a tarball, removing a path prefix
+
+[![flate2-badge]][flate2] [![tar-badge]][tar] [![cat-compression-badge]][cat-compression]
+
+Iterates through the elements of a tarball, removing their path prefix.
+
+Creates an archive, `archive.tar.gz`, using the approach shown in the
+[compression][ex-tar-compress] example and prefixing the archive contents with the
+path `bundle/logs`. After creating a gzip compressed archive, iterate through the entries
+by creating a [`tar::Archive`] and calling its [`tar::Archive::entries`] method.
+The prefix `bundle/logs` is removed from the paths using
+[`std::path::Path::strip_prefix`] prior to printing to stdout. 
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+extern crate tar;
+extern crate flate2;
+#
+# error_chain! {
+#     foreign_links {
+#         Io(::std::io::Error);
+#         StripPrefixError(::std::path::StripPrefixError);
+#     }
+# }
+
+use flate2::Compression;
+use flate2::read::GzDecoder;
+use flate2::write::GzEncoder;
+use std::fs::File;
+use std::path::Path;
+use tar::Archive;
+
+fn run() -> Result<()> {
+    // Create tarball archive
```

Comment:
no need to create the tarball here lets assume it exists. the example is `no_run` anyway so our testrunner will only verify that it compiles.